Moves Automatic Import to solution-agnostic section (#5889)
elastic/docs-content: explore-analyze/ai-features/automatic-import.md

Reviewer: I'd strongly suggest to change the majority of admonitions:
- Requirements can be a sub-section
- Notes on sample data can be a sub-section (Prepare your sample data for example)
- Recommended LLM models can be a sub-section, with the important note.

---

> navigation_title: Automatic import

Reviewer 2: Should the `nav_title` match the case of `Automatic Import` like everything else?

description: Use Automatic Import with an LLM to build a custom Elastic integration from a data sample when no prebuilt integration exists, for Security, Observability, and other solutions.
mapped_pages:
  - https://www.elastic.co/guide/en/security/current/automatic-import.html
  - https://www.elastic.co/guide/en/serverless/current/security-automatic-import.html
applies_to:
  stack: ga
  serverless:
    security: ga
    observability: ga
products:
  - id: kibana
  - id: cloud-serverless
  - id: security
  - id: observability
---

> # Automatic import [automatic-import]

Reviewer 2: Same case comment as above. Is it `Automatic import` or `Automatic Import`?

---
navigation_title: Automatic import
description: Use Automatic Import with an LLM to build a custom Elastic integration from a data sample when no prebuilt integration exists, for Security, Observability, and other solutions.
mapped_pages:
  - https://www.elastic.co/guide/en/security/current/automatic-import.html
  - https://www.elastic.co/guide/en/serverless/current/security-automatic-import.html
applies_to:
  stack: ga
  serverless:
    security: ga
    observability: ga
products:
  - id: kibana
  - id: cloud-serverless
  - id: security
  - id: observability
---

# Automatic import [automatic-import]

Automatic Import helps you quickly parse, ingest, and create [ECS mappings](https://www.elastic.co/elasticsearch/common-schema) for data from sources that don’t yet have prebuilt Elastic integrations. You can use it with {{elastic-sec}}, {{observability}}, and other Elastic solutions that rely on {{agent}} and integrations, so you can cover custom or niche data sources without building a full integration by hand. Automatic Import uses a large language model (LLM) with specialized instructions to analyze your source data and create a custom integration.

> While Elastic has 400+ [prebuilt data integrations](https://docs.elastic.co/en/integrations), Automatic Import helps you extend data coverage when you need logs or events from technologies that don’t have a dedicated integration yet. Elastic integrations (including those created by Automatic Import) normalize data to [the Elastic Common Schema (ECS)](ecs://reference/index.md), which creates uniformity across dashboards, search, alerts, machine learning, and more.

Reviewer 3: ```suggestion
Automatic Import parses, ingests, and maps data to [ECS](https://www.elastic.co/elasticsearch/common-schema) for sources that don’t have prebuilt Elastic integrations. It works with {{elastic-sec}}, {{observability}}, and other solutions that rely on {{agent}} and integrations. This lets you onboard custom or niche data sources without building a full integration manually.

Automatic Import uses a large language model (LLM) with specialized instructions to analyze source data and generate a custom integration.

Elastic integrations, including those created by Automatic Import, normalize data to [the Elastic Common Schema (ECS)](ecs://reference/index.md). This standardization provides consistent use across dashboards, search, alerts, and machine learning features.

Refer to [prebuilt data integrations](https://docs.elastic.co/en/integrations) for a full list of Elastic's 400+ integrations.
```

I can't pinpoint exactly why, but your intro seems a bit marketing-y. This is just my suggestion how to make it more direct/fact-oriented. Take it, leave it, mix it with your version 🤷‍♀️

::::{tip}
See an [interactive demo of Automatic Import](https://elastic.navattic.com/automatic-import) before you set it up in your environment.
::::

::::{admonition} Requirements

* A working [LLM connector](/explore-analyze/ai-features/llm-guides/llm-connectors.md).
* {{stack}} users: An [Enterprise](https://www.elastic.co/pricing) subscription.
* {{serverless-short}} **{{sec-serverless}}** projects: the [Security Analytics Complete](/deploy-manage/deploy/elastic-cloud/project-settings.md#elastic-sec-project-features) feature tier.
* {{serverless-short}} **{{obs-serverless}}** projects: the [Observability Complete](/deploy-manage/deploy/elastic-cloud/project-settings.md#obs-serverless-project-features) feature tier.
* A sample of the data you want to import.

::::

::::{admonition} Notes on sample data
To use Automatic Import, you must provide a sample of the data you wish to import. An LLM will process that sample and automatically create an integration suitable for processing the data represented by the sample. **Automatic Import supports the following sample formats: JSON, NDJSON, CSV, and syslog (structured and unstructured).**

> * You can upload a sample of arbitrary size. The LLM will detect its format and select up to 100 documents for detailed analysis.

Reviewer 3: ```suggestion
* You can upload a sample of any size. The LLM detects its format and selects up to 100 documents for detailed analysis.
```

* The more variety in your sample, the more accurate the pipeline will be. For best results, include a wide range of unique log entries in your sample instead of repeating similar logs.

> * When uploading a CSV, a header with column names will be automatically recognized. However if the header is not present, the LLM will still attempt to create descriptive field names based on field formats and values.

Reviewer 3: ```suggestion
* When uploading a CSV, a header with column names are automatically recognized. If the header is not present, the LLM attempts to generate descriptive field names based on field formats and values.
```

Avoids future tense. BTW, you have several more instances where you use future tense which could be rewritten in present tense.

* For JSON and NDJSON samples, each object in your sample should represent an event, and you should avoid deeply nested object structures.

> * When you select `API (CEL input)` as one of the sources, you will be prompted to provide the associated OpenAPI specification (OAS) file to generate a CEL program that consumes this API.

Reviewer: CEL term is unclear. Maybe a reference would be helpful.

We can say Common Expression Language (CEL) the first time it appears, and also consider linking to:

- The [filebeat CEL input](https://www.elastic.co/docs/reference/beats/filebeat/filebeat-input-cel)
- [CEL project](https://github.com/google/cel-spec)

4. Select an [LLM connector](/explore-analyze/ai-features/llm-guides/llm-connectors.md).
5. Define how your new integration will appear on the Integrations page by providing a **Title**, **Description**, and **Logo**.  Click **Next**.
6. Define your integration’s package name, which will prefix the imported event fields.
7. Define your **Data stream title**, **Data stream description**, and **Data stream name**. These fields appear on the integration’s configuration page to help identify the data stream it writes to.
8. Select your [**Data collection method**](beats://reference/filebeat/configuration-filebeat-options.md). This determines how your new integration will ingest the data (for example, from an S3 bucket, an HTTP endpoint, or a file stream).

   ::::{admonition} Importing CEL data
   :class: note
   If you select **API (CEL input)**, you’ll have the additional option to upload the API’s OAS file here. After you do, the LLM will use it to determine which API endpoints (GET only), query parameters, and data structures to use in the new custom integration. You will then select which API endpoints to consume and your authentication method before uploading your sample data.

>    ::::

Reviewer: A direct link to the CEL input might be interesting there.
